libs/api-implementation/regression-tests/src/lib/regression-tests.controller.ts

```diff
@@ -0,0 +1,26 @@
+import { Controller, Post, Body, Get } from '@nestjs/common';
+import { RegressionTestService } from './services/regression-tests.service';
+import { RegressionTestDTO } from './dto/regression-test.dto';
+
+@Controller('regression-test')
+export class RegressiontTestController {
+  constructor(private service: RegressionTestService) {}
+
+  @Post()
+  async createRegressionTest(@Body() regressionTestDTO: RegressionTestDTO) {
+    try {
+      return await this.service.create(regressionTestDTO.command)
+    } catch (error) {
+      return `Error creating the regression test`
+    }
```

Comment:
We could use the improvement in error handling here as well, what do you think?

Reply:
@zelzhan Any comments here?

Reply:
I think we misunderstand each other. This is the controller, and I'm logging error in the service. 

Sorry if I didn't get you right. If I didn't get you, could you please elaborate more? Thank you!

Reply:
Oh, right. I want to understand how is it different when you are catching an error in the controller and in the service? If I understand well, will it be more convenient to use the same method of logging when we catch an error (i.e. console.debug with trace followed by console.error with a message) independent of the file it is in?

Reply:
I believe that error check in service is good practice. The controller should not be aware of the data access layer. The only reason why I put catch block in the controller: so that client will be sure that something went wrong. Is that make sense?

Reply:
Right. That makes a lot of sense, thank you for taking time to explain and make me understand, really appreciate it :tada: